R/zchunk_L241.trade_input.R

```diff
@@ -0,0 +1,194 @@
+    L241.low_reg <- L241.pcgdp_2010$region[L241.pcgdp_2010$value < aglu.LOW_GROWTH_PCGDP]
+
+    # Create table of regions, technologies, and all base years (47-51)
+    # NOTE: Easiest if the model base years are subsetted from a full table as a last step in the construction of each of these tables
+    A_demand_technology %>%
+      write_to_all_regions(c(aglu.NAMES_TECH, "minicam.energy.input", "market.name"), GCAM_region_names) %>%
+      mutate(stub.technology = technology) ->
+      A_demand_technology_R
+    A_an_input_technology %>%
+      write_to_all_regions(c(aglu.NAMES_TECH, "minicam.energy.input", "market.name"), GCAM_region_names) %>%
+      mutate(stub.technology = technology) ->
+      A_an_input_technology_R
+
+    # Add lookup vectors to level1 output tables (55-59)
+    # Add region names to Level1 data tables
+    aglu_demand_calyears <- HISTORICAL_YEARS[HISTORICAL_YEARS %in% MODEL_YEARS]
+    aglu_demand_futureyears <- MODEL_YEARS[!MODEL_YEARS %in% aglu_demand_calyears]
+    L101.ag_kcalg_R_C_Y %>%
+      filter(year %in% aglu_demand_calyears) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      filter(year == max(aglu_demand_calyears)) %>%
+      select(region, GCAM_commodity, value) ->
+      L241.ag_kcalg_R_C_Yf.mlt
+
+    # Fraction of food that should be produced locally
+    LOCAL_FOOD_FRACT <- 0.8
+
+    # Coefficient for ag_trade of food crops (incl secondary products) (61-69)
+    A_demand_technology_R %>%
+      filter(supplysector == "FoodDemand_Crops") %>%
+      left_join_error_no_match(L241.ag_kcalg_R_C_Yf.mlt, by = c("region" = "region", "technology" = "GCAM_commodity")) %>%
+      mutate(coefficient = LOCAL_FOOD_FRACT / round(value, aglu.DIGITS_CALOUTPUT),
+             minicam.energy.input = "ag_trade") %>%
+      select(-value) %>%
+      repeat_add_columns(tibble(year = aglu_demand_futureyears)) %>%
+      select(one_of(aglu.NAMES_STUBTECHCOEF_NM)) ->
+      L241.StubAgTradeCoeff_food
+
+    # Coefficient for ag_trade of non-food crops (incl secondary products) (71-76)
+    A_demand_technology_R %>%
+      filter(supplysector == "NonFoodDemand_Crops") %>%
+      mutate(coefficient = LOCAL_FOOD_FRACT,
+             minicam.energy.input = "ag_trade") %>%
+      repeat_add_columns(tibble(year = aglu_demand_futureyears)) %>%
+      select(one_of(aglu.NAMES_STUBTECHCOEF_NM)) ->
+      L241.StubAgTradeCoeff_nonfood
+
+    # Coefficient for ag_trade of feed crops (incl secondary products) (78-83)
+    A_an_input_technology_R %>%
+      filter(supplysector == "FeedCrops") %>%
+      mutate(coefficient = LOCAL_FOOD_FRACT,
+             minicam.energy.input = "ag_trade") %>%
+      repeat_add_columns(tibble(year = aglu_demand_futureyears)) %>%
+      select(one_of(aglu.NAMES_STUBTECHCOEF_NM)) ->
+      L241.StubAgTradeCoeff_feed
+
+    # Output coefficient for RES ag_trade of food crops (incl secondary products) (85-90)
```

Comment:
This chunk sets up the supply of `ag_trade` credits. It says that each supply sector generates 1 credit for each unit of crop/livestock produced.